crypto/x509/x509_lu.c

```diff
@@ -613,13 +615,21 @@ int X509_STORE_CTX_get1_issuer(X509 **issuer, X509_STORE_CTX *ctx, X509 *x) {
       }
       if (ctx->check_issued(ctx, x, pobj->data.x509)) {
         *issuer = pobj->data.x509;
-        X509_OBJECT_up_ref_count(pobj);
         ret = 1;
```

Comment:
we consider the lookup successful even if all of the issuer certs are expired?

Reply:
Yeah the reason being that `X509_STORE_CTX_get1_issuer` is mainly used to find a matching issuer. The cert may be expired, but it's still considered a match.
This aligns with what OpenSSL1.1.1 has: https://github.com/openssl/openssl/blob/OpenSSL_1_1_1-stable/crypto/x509/x509_lu.c#L724-L735

Reply:
Your comments led me to take a closer look at `X509_STORE_CTX_get1_issuer` and I realized that this hadn't been properly tested with my initial set up. I added some tests to specifically test the new function behavior!